worlds/mm2/__init__.py

```diff
@@ -0,0 +1,312 @@
+        item = item_table[name]
+        classification = ItemClassification.filler
+        if item.progression and not force_non_progression:
+            classification = ItemClassification.progression_skip_balancing \
+                if item.skip_balancing else ItemClassification.progression
+        if item.useful:
+            classification |= ItemClassification.useful
+        return MM2Item(name, classification, item.code, self.player)
+
+    def get_filler_item_name(self) -> str:
+        return self.multiworld.random.choices(list(filler_item_weights.keys()),
+                                              weights=list(filler_item_weights.values()))[0]
+
+    def create_items(self) -> None:
+        itempool = []
+        # grab first robot master
+        robot_master = self.item_id_to_name[0x880101 + self.options.starting_robot_master.value]
+        self.multiworld.push_precollected(self.create_item(robot_master))
+        itempool.extend([self.create_item(name) for name in stage_access_table.keys()
+                         if name != robot_master])
+        itempool.extend([self.create_item(name) for name in robot_master_weapon_table.keys()])
+        itempool.extend([self.create_item(name) for name in item_item_table.keys()])
+        total_checks = 24
+        if self.options.consumables in (self.options.consumables.option_1up_etank,
+                                        self.options.consumables.option_all):
+            total_checks += 20
+        if self.options.consumables in (self.options.consumables.option_weapon_health,
+                                        self.options.consumables.option_all):
+            total_checks += 27
+        remaining = total_checks - len(itempool)
+        itempool.extend([self.create_item(name)
+                         for name in self.multiworld.random.choices(list(filler_item_weights.keys()),
+                                                                    weights=list(filler_item_weights.values()),
+                                                                    k=remaining)])
+        self.multiworld.itempool += itempool
+
+    set_rules = set_rules
+
+    def generate_early(self) -> None:
+        if (not self.options.yoku_jumps
+            and self.options.starting_robot_master.current_key == "heat_man") or \
+                (not self.options.enable_lasers
+                 and self.options.starting_robot_master.current_key == "quick_man"):
+            robot_master_pool = [1, 2, 3, 5, 6, 7, ]
+            if self.options.yoku_jumps:
+                robot_master_pool.append(0)
+            if self.options.enable_lasers:
+                robot_master_pool.append(4)
+            self.options.starting_robot_master.value = self.random.choice(robot_master_pool)
+            logger.warning(
+                f"Incompatible starting Robot Master, changing to "
+                f"{self.options.starting_robot_master.current_key.replace('_', ' ').title()}")
```

Comment:
Currently, if the player chooses random, it will give this warning sometimes. It could be made to not do this, or just be accepted as-is -- I don't really care either way, just want to make sure it's known.

Also, might be a good idea to throw the slot name in here too.

Reply:
Mentioned my stance on this back during the discussion in #ap-world-dev, I much prefer being loud about the error even if it was random that chose it and not the player specifically. Much simpler to handle anyways, handling it otherwise involves overriding from_text.